changelog/0037-upgrade-to-elasticsearch7.yml

```
@@ -0,0 +1,350 @@
+            cchq <env> service pillowtop stop --only user-pillow
+          ```
+  3. At this point, your ES2 hosts should not be getting any HTTP requests.
+      - You can confirm this by doing a tcpdump on Elasticsearch port on any of your Elasticsearch 2 hosts
+          ```
+          tcpdump -A -s 0 'tcp port 9200 and (((ip[2:2] - ((ip[0]&0xf)<<2)) - ((tcp[12]&0xf0)>>2)) != 0)'_
+          ```
+  4. Reindex by running below command for each of the index in`['forms', 'cases', 'users', 'domains', 'apps', 'groups', 'sms', 'report_cases', 'report_xforms', 'case_search']`
+      ```
+      ./manage.py reindex_es_from_remote <index_name> <es2_host>
+      ```
+      - These can be run concurrently in different tmux/screen sessions
+      - Note that these management commands are to be run from the release that was created in Setup phase, which is pointing to Elasticsearch 7
+  5. At this point, all of the Elasticsearch 2 data should be in Elasticsearch 7.
+      - You can confirm this by running this command to display the number of documents in both ES clusters:
+          ```
+          ./manage.py reindex_es_from_remote --print_index_size
+          ```
+      - Alternatively you may use an ES monitoring tool such as https://www.elastichq.org/
+  6. Run below command to update Elasticsearch settings on ES7 indices for usage
+      ```
+      ./manage.py initialize_es_indices --set-for-usage
+      ```
+  7. Update HQ config to point to Elasticsearch 7.
+      - Update below settings in you env's public.yml in addition to other setting changes made in git-upgrade-branch in the Setup phase
+          ```
+          ELASTICSEARCH_MAJOR_VERSION: 7
+          ```
+      - Update config
+          ```
+          cchq <env> update-config --branch=<git-upgrade-branch>
+          ```
+      - Merge the git-upgrade-branch into master  
+  9. Make sure that HQ is able to connect to Elasticsearch 7
+      ```
+      cchq <env> django-manage check_services
+      ```
+  10. End downtime
+      ```
+      cchq <env> downtime end
+      ```
+  That's it, you are running HQ on Elasticsearch 7!
+
+
+
+  ### Reindex and Switchover separately
+
+  If your form/case index sizes are large, you can follow this method to first reindex without downtime and then switchover with downtime.
+
+  #### Reindex
+
+  Reindex phase involves reindexing data from Elasticsearch 2 cluster into new Elasticsearch 7 cluster. Form and Case indices can be indexed in background while the HQ application continues to be live. This reindex can be performed based on the `inserted_at` date field up to a day before switchover.  Using this parameter multiple reindex processes can be run concurrently for different date ranges enabling you to reindex very large indices. The smaller offset data inserted newly during and after the reindex phase can be reindexed  during the last phase of switchover with downtime. This enables minimising the overall downtime required to upgrade drastically. 
```

split over multiple lines
```suggestion
  Reindex phase involves reindexing data from Elasticsearch 2 cluster into new Elasticsearch 7 cluster.
  Form and Case indices can be indexed in the background while the HQ application continues to be live.
  This reindex can be performed based on the `inserted_at` date field up to a day before switchover.  Using this
  parameter multiple reindex processes can be run concurrently for different date ranges enabling you to reindex very
  large indices. The data inserted during and after the reindex phase can be reindexed during the last phase of
  switchover with downtime. This enables minimising the overall downtime required to upgrade drastically. 
```